File: week 2/p1 유쾌한 점퍼 (Jolly Jumpers)/Main.java
import java.util.Scanner;

class Main {
	public static void main(String[] args) throws Exception{
		
		Scanner input = new Scanner(System.in);
		String line;
		int n;
		int prev, next;
		int i;
		boolean jollyTorF;
		
		while (input.hasNextLine()) {
			line = input.nextLine();
			n = Integer.parseInt(line.split(" ")[0]);
			boolean[] jollyArray = new boolean[n];
			prev = Integer.parseInt(line.split(" ")[1]);
			
			for (i = 2; i <= n; i++) {
				next = Integer.parseInt(line.split(" ")[i]);
				if (Math.abs(prev - next) < n && Math.abs(prev - next) > 0)
					jollyArray[Math.abs(prev - next)] = true;
				prev = next;
			}
			
			jollyTorF = true;
			for (i = 1; i < n; i++) {
				if (!jollyArray[i]) {
					jollyTorF = false;
					break;
				}
			}
			
			if (jollyTorF)
				System.out.println("Jolly");
			else
				System.out.println("Not jolly");
		}
		
		input.close();
	}
}

File: week 2/p1 유쾌한 점퍼 (Jolly Jumpers)/Main.py
# 입력받음
while True:
	try:
		input_list = input().split()
		n = int(input_list[0])
		n_list = list(map(int, input_list[1:]))
		jumper_list = []

		# 앞뒤에 있는 숫자 차의 절대값을 구함
		for i in range(n-1):
			jumper_list.append(abs(n_list[i] - n_list[i+1]))

		# Jolly인지 Not jolly인지 확인
		isJolly = True
		for jumper in range(len(jumper_list)):
			if jumper+1 not in jumper_list:
				print("Not jolly")
				isJolly = False
				break
		if isJolly:
			print("Jolly")
	except EOFError:
		break